### PR TITLE
add error handling for missing coupling method

### LINKED ISSUE #1119: esm_runscripts version 6.19.3 giving error message on aleph since the New Year
Since this year, the esm_runscripts version 6.19.3 is giving an error message no matter if I want to start a completely new simulation, branch off from an existing simulation, or restarting an existing simulation to add more simulated years. 

After submitting 

awiiccp5@elogin1:/proj/awiiccp5:> esm_runscripts awicm3-frontiers-xios-aleph-TCO95L91-HR_generateweights.yaml --open-run -e test

I am getting this error message:

STARTING SIMULATION JOB!
Experiment ID = test
Setup = awicm3
This setup consists of:
- fesom
- oifs
- rnfmap
- oasis3mct
- xios
Experiment is installed in:
       /scratch/awiiccp5//test
================================================================================

Traceback (most recent call last):
  File "/home/awiiccp5/miniconda3/bin/esm_runscripts", line 33, in <module>
    sys.exit(load_entry_point('esm-tools', 'console_scripts', 'esm_runscripts')())
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/cli.py", line 278, in main
    Setup()
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/sim_objects.py", line 69, in __call__
    self.prepcompute()
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/sim_objects.py", line 161, in prepcompute
    self.config = prepcompute.run_job(self.config)
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/prepcompute.py", line 32, in run_job
    config = evaluate(config, "prepcompute", "prepcompute_recipe")
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/helpers.py", line 69, in evaluate
    config = esm_plugin_manager.work_through_recipe(
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_plugin_manager/esm_plugin_manager.py", line 141, in work_through_recipe
    config = getattr(submodule, workitem)(config)
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/prepcompute.py", line 91, in prepare_coupler_files
    coupler_filename = config["general"]["coupler"].prepare(
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/coupler.py", line 56, in prepare
    self.add_couplings(full_config)
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/coupler.py", line 370, in add_couplings
    self.coupler.add_coupling(
  File "/mnt/lustre/home/awiiccp5/esm_tools/src/esm_runscripts/oasis.py", line 128, in add_coupling
    alltimes = transformation.get("time_transformation", "bla")
AttributeError: 'NoneType' object has no attribute 'get'

This happens no matter which (previously) functioning run-script I would submit from this directory.

It is taking the esm_tools version from /home/awiiccp5/esm_tools and the esm_runscripts executable from /home/awiiccp5/miniconda3/bin

### PR DESCRIPTION
**Pull request by mandresm**
Adds error handling for issue #1119

Closes #1119

### CODE REVIEW

**Comment by mandresm** [comment]
Screenshot 2024-01-18 at 17 04 05

File: src/esm_runscripts/coupler.py [hunk 1]
```diff
@@ -168,6 +170,13 @@ def add_files(self, full_config):
                             transf_info = full_config[self.name]["coupling_methods"][
                                 interpolation
                             ]
+                        else:
+                            esm_parser.user_error(
+                                "Missing coupling method",
+                                f"The coupling method ``{interpolation}`` defined in "
+                                f"the ``{self.name}.coupling_target_fields`` is not "
+                                f"defined anywhere in ``{self.name}.coupling_methods``"
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                                f"defined anywhere in ``{self.name}.coupling_methods``."
```

**Review by pgierz — APPROVED** [review]

File: src/esm_runscripts/coupler.py (hunk 1 above)
Comment (mandresm) [reply src/esm_runscripts/coupler.py]: Thanks!

**Comment by mandresm** [comment]
#bump

**Pull request merged**